add profile name to post and comment APIs

aliceriot opened the pull request:
#### What are the relevant tickets?

#188 

#### What's this PR do?

this adds the `.name` from the user's profile to the results returned by both the comment and post APIs, and makes the changes in the UI to display that information.

#### How should this be manually tested?

If you look at a post or a channel you should see names instead of usernames, same with comments on a post.

rhysyngsun commented:
I think this PR needs to be rebased/refactored off @noisecapella's PR #198 once that merges so that we can accoutn for cases where `User` doesn't exist.

aliceriot commented:
@rhysyngsun ok I have rebased over George's changes and incorporated the newer pattern for including profile information on the responses. I also added `.select_related('profile')` when we grab the users, to avoid making another query on each call to the serializer (for the post and comment list views) to grab the profile information.

rhysyngsun commented on channels/serializers.py:
```diff
@@ -144,6 +145,13 @@ def get_url(self, instance):
+        user = self._get_user(instance)
+        if user and user.profile.name:
+            return user.profile.name
+        return "Someone"
```
Could you change this to be "[deleted]" to preserve the original default value?

aliceriot replied:
sure

rhysyngsun commented on channels/serializers.py:
```diff
@@ -290,6 +299,15 @@ def get_profile_image(self, instance):
+
+        if user and user.profile.name:
+            return user.profile.name
+        return "Someone"
```
Same comment as above

rhysyngsun commented on static/js/components/PostDisplay.js:
```diff
@@ -115,7 +115,7 @@ class PostDisplay extends React.Component {
           </div>
           <div className="authored-by">
-            <a>{post.author_id || "someone"}</a> {formattedDate}{" "}
+            <a>{post.author_name || "someone"}</a> {formattedDate}{" "}
```
This will never be null now, so we don't need the default.

rhysyngsun commented:
Functionality looks good.

aliceriot commented:
@rhysyngsun I believe I address all your comments
